src/protocols/ethereum/codegen/abi.js

```diff
@@ -208,6 +225,54 @@ module.exports = class AbiCodeGenerator {
             `return new ${paramsClassName}(this)`,
           ),
         )
+
+        // Fixture generation
+        try {
```

Comment:
I'm trying to think here if there isn't a better way to put this here.

I'm not sure I like having to do synchronous operations + big conditions here for the fixtures.

Perhaps we can go forward with this so I don't block you and then I can think of a better way to architect this so that it isn't so intrusive to the regular `abi` code generation.